frontend/src/js/components/subscription/SubscriptionPage.tsx

```diff
@@ -0,0 +1,353 @@
+
+import { SupportLink } from '@northern.tech/common-ui/SupportLink';
+import { ADDONS, Addon, AvailableAddon, AvailablePlans, PLANS, Plan } from '@northern.tech/store/appSlice/constants';
+import { getStripeKey } from '@northern.tech/store/appSlice/selectors';
+import { TIMEOUTS } from '@northern.tech/store/commonConstants';
+import { getDeviceLimit } from '@northern.tech/store/devicesSlice/selectors';
+import { getOrganization } from '@northern.tech/store/organizationSlice/selectors';
+import { getBillingPreview, getCurrentCard, getUserBilling, requestPlanChange } from '@northern.tech/store/organizationSlice/thunks';
+import { useAppDispatch } from '@northern.tech/store/store';
+import { useDebounce } from '@northern.tech/utils/debouncehook';
+import { Elements } from '@stripe/react-stripe-js';
+
+import { addOnsToString } from '../settings/Upgrade';
+import { SubscriptionAddon } from './SubscriptionAddon';
+import { SubscriptionDrawer } from './SubscriptionDrawer';
+import { SubscriptionSummary } from './SubscriptionSummary';
+import { parseSubscriptionPreview } from './utils';
+
+let stripePromise = null;
+export type PreviewPrice = { addons: { [key in AvailableAddon]: number }; plan: number; total: number };
+
+const DIVISIBILITY_STEP = 50;
+const enterpriseDeviceCount = PLANS.enterprise.minimalDeviceCount;
+const enterpriseRequestPlaceholder = 'Tell us a little about your requirements and device fleet size, so we can provide you with an accurate quote';
+type SelectedAddons = { [key in AvailableAddon]: boolean };
+export const SubscriptionPage = () => {
+  const [selectedPlan, setSelectedPlan] = useState<Plan>(PLANS.os);
+  const [selectedAddons, setSelectedAddons] = useState<SelectedAddons>({ configure: false, monitor: false, troubleshoot: false });
+  const [previewPrice, setPreviewPrice] = useState<PreviewPrice>();
+  const [order, setOrder] = useState();
+  const [contactReason, setContactReason] = useState<string>('');
+  const [inputHelperText, setInputHelperText] = useState<string>(`The minimum limit for ${selectedPlan.name} is ${selectedPlan.minimalDeviceCount}`);
+  const [limit, setLimit] = useState<number>(selectedPlan.minimalDeviceCount);
+  const [enterpriseMessage, setEnterpriseMessage] = useState('');
+  const stripeAPIKey = useSelector(getStripeKey);
+  const [isPreviewLoading, setIsPreviewLoading] = useState(false);
+  const [showUpgradeDrawer, setShowUpgradeDrawer] = useState(false);
+  const [loadingFinished, setLoadingFinished] = useState(!stripeAPIKey);
+
+  const dispatch = useAppDispatch();
+  const currentDeviceLimit = useSelector(getDeviceLimit);
+  const org = useSelector(getOrganization);
+  const { addons: orgAddOns = [], plan: currentPlan = PLANS.os.id as AvailablePlans, trial: isTrial = true, isLoaded: isOrgLoaded } = org;
+  const plan = Object.values(PLANS).find(plan => plan.id === (isTrial ? PLANS.os.id : currentPlan));
```

Comment:
```suggestion
  const plan = Object.values(PLANS).find(plan => plan.id === (isTrial ? PLANS.os.id : currentPlan)) || PLANS.os;
```
to be in line with the `currentPlan` assignment above and to avoid the need for the repeated checks for a set plan further down?